### DISCUSSION TITLE
Pre-install questions

### DISCUSSION THREAD

**Discussion by lucianobosco** [spatie/laravel-medialibrary#2209] [General] [answered]
Hi! I have a few questions that I'd like to clarify before integrating this package into my project.

1. I saw that Spatie also has an [image-optimizer](https://github.com/spatie/image-optimizer) package. I'm wondering if this package covers the same functionalities for optimizing images or I should use both and chain functionalities in some way. I'm aware that this package can optimize, but not sure if they have the same efficiency.
2. Since I need to create a Media CRUD by allowing users to directly upload medias in no context, can media models be created without being attached to any other model? Something like `Media::create($pathToFile)`
3. Does this package allow me to associate videos, audios, and files such as .docx, .xlsx, .pdfs?
4. Can I use this package to attach and upload medias directly to S3 from POST submission?
5. Will I be able to customize the media DB table before running package migrations for custom properties? Or this must be handled post installation?

Thanks in advance

**Comment by freekmurze** [comment, marked as answer]
1. Media Library uses that image optimiser under the hood
2. No, for that behaviour you'll need a dummy model, our use https://medialibrary.pro
3. Yes
4. No, the components offered in https://medialibrary.pro can do this
5. You can customise it anytime you want. We do offer custom properties that allow you to add any custom properties without have to change the db: https://spatie.be/docs/laravel-medialibrary/v9/advanced-usage/using-custom-properties
